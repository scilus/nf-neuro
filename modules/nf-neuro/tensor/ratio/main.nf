process TENSOR_RATIO {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_2.0.2.sif':
        'scilus/scilus:latest' }"

    input:
        tuple val(meta), path(image)

    output:
        tuple val(meta), path("*_Dxx_to_Dyy.nii.gz")        , emit: tensor_x_to_y
        tuple val(meta), path("*_Dxx_to_Dzz.nii.gz")        , emit: tensor_x_to_z
        path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    fslsplit $image
    fslmaths ${image}_0000.nii.gz -div ${image}_0003.nii.gz ${prefix}_Dxx_to_Dyy.nii.gz
    fslmaths ${image}_0000.nii.gz -div ${image}_0005.nii.gz ${prefix}_Dxx_to_Dzz.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fsl: \$(fslsplit -version 2>&1 | sed -n 's/FSLSPLIT version \\([0-9.]\\+\\)/\\1/p')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_Dxx_to_Dyy.nii.gz
    touch ${prefix}_Dxx_to_Dzz.nii.gz
    fslsplit
    fslmaths

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        fsl: \$(fslsplit -version 2>&1 | sed -n 's/FSLSPLIT version \\([0-9.]\\+\\)/\\1/p')
    END_VERSIONS
    """
}
