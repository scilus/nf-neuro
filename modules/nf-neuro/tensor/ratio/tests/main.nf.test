// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test tensor/ratio
nextflow_process {

    name "Test Process TENSOR_RATIO"
    script "../main.nf"
    process "TENSOR_RATIO"

    tag "modules"
    tag "modules_nfcore"
    tag "tensor"
    tag "tensor/ratio"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "dti.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }
    test("tensorratio - classic") {
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dti/tensor.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("tensorratio - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dti/tensor.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
