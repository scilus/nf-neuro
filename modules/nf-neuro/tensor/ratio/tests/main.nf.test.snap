{
    "tensorratio - classic": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "tensor_x_to_y": [
                    
                ],
                "tensor_x_to_z": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-27T14:05:10.553221131"
    },
    "tensorratio - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "tensor_x_to_y": [
                    
                ],
                "tensor_x_to_z": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-27T14:05:28.311339916"
    }
}